config/runtimes/kserve-huggingfaceserver-multinode.yaml

```diff
@@ -0,0 +1,231 @@
+        limits:
+          cpu: "16"
+          memory: 48Gi
+          nvidia.com/gpu: "1"
+        requests:
+          cpu: "8"
+      volumeMounts:
+        - name: shm
+          mountPath: /dev/shm     
+      livenessProbe:
+        failureThreshold: 3
+        periodSeconds: 30
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |            
+              # Check GPU usage (If worker node restarted, it becomes NOT ready)
+              gpu_status=$(ray status | grep GPU)
+              used_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f1)
+              reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+              # Determine health status based on GPU usage
+              if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                  echo "Liveness Probe: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                  exit 1
+              else
+                  exit 0                  
+              fi 
+      readinessProbe:
+        failureThreshold: 2
+        periodSeconds: 10
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |
+              # Check if the registered ray nodes count is the same as PIPELINE_PARALLEL_SIZE
+              if [[ $(ray status | grep -c node_) -eq ${PIPELINE_PARALLEL_SIZE} ]]; then
+                exit 0
+              else
+                exit 1
+              fi &
+              p1=$!
+
+              # Check if huggingface server health
+              curl --silent --max-time 5 http://localhost:8080 &
+              p2=$!
+
+              # Check GPU usage (If worker node restarted, it becomes NOT ready)
+              gpu_status=$(ray status | grep GPU)
+              used_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f1)
+              reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+              # Determine health status based on GPU usage
+              if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                  echo "Readiness Probe: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                  exit 1
+              else
+                  exit 0                  
+              fi & # Run this check in the background
+              p3=$!
+
+              # Wait for all processes to complete
+              wait $p1
+              status1=$?
+              wait $p2
+              status2=$?
+              wait $p3
+              status3=$?
+
+              # Check all return codes
+              if [[ $status1 -ne 0 || $status2 -ne 0 || $status3 -ne 0 ]]; then
+                  echo "One or more processes failed."
+                  exit 1
+              else
+                  echo "All processes succeeded."
+                  exit 0
+              fi
+      startupProbe:
+        failureThreshold: 40
```

Comment:
Better to configure initial delay seconds if we know this is going to take a while

Reply:
The `ray status` command should return immediately when the pod is starting up, but I have added a safety delay of 5 seconds as a precaution.